### ISSUE TITLE
[python] predict(..., ntree_limit=?, pred_leaf=True) - ntree_limit  issue 

### ISSUE THREAD

**Issue by mandarup** [dmlc/xgboost#2209]
the shape of output from `predict()` is different from `ntree_limit` option . 

Following the definition of parameters:

        ntree_limit : int
            Limit number of trees in the prediction; defaults to 0 (use all trees).

        pred_leaf : bool
            When this option is on, the output will be a matrix of (nsample, ntrees)
            with each record indicating the predicted leaf index of each sample in each tree.
            Note that the leaf index of a tree is unique per tree, so you may find leaf 1
            in both tree 1 and tree 0.

My understanding is that the output from `predict()` should be of shape  `[data.shape[0], ntree_limit]`.  Am I misunderstanding what `ntree_limit`  is supposed to do or is this a bug? 

here is a reproducible example using ``xgboost/demo/multiclass_classification``  

    import numpy as np
    import xgboost as xgb
    import pandas as pd

    # label need to be 0 to num_class -1
    data = np.loadtxt('./dermatology.data', delimiter=',',converters={33: lambda x:int(x == '?'), 34: lambda x:int(x)-1 } )
    sz = data.shape

    train = data[:int(sz[0] * 0.7), :]
    test = data[int(sz[0] * 0.7):, :]

    train_X = train[:,0:33]
    train_Y = train[:, 34]

    test_X = test[:,0:33]
    test_Y = test[:, 34]

    xg_train = xgb.DMatrix( train_X, label=train_Y)
    xg_test = xgb.DMatrix(test_X, label=test_Y)
    # setup parameters for xgboost
    param = {}
    # use softmax multi-class classification
    param['objective'] = 'multi:softmax'
    # scale weight of positive examples
    param['eta'] = 0.1
    param['max_depth'] = 6
    param['silent'] = 1
    param['nthread'] = 4
    param['num_class'] = 6

    watchlist = [ (xg_train,'train'), (xg_test, 'test') ]
    num_round = 5
    bst = xgb.train(param, xg_train, num_round, watchlist );

    # get prediction
    pred = bst.predict( xg_test );

    print ('predicting, classification error=%f'
        % (sum( int(pred[i]) != test_Y[i] for i in range(len(test_Y))) / float(len(test_Y)) ))

    # Leaf info : 1 tree
    pred_leaf_1 = bst.predict( xg_test, output_margin=False, ntree_limit=1, pred_leaf=True)

    print(pred_leaf_1.shape)
    # >>> (110, 6)

     # Leaf info : 2 trees
     pred_leaf_2 = bst.predict( xg_test, output_margin=False, ntree_limit=2, pred_leaf=True)
     print(pred_leaf_2.shape)
     # >>> (110, 12)

## Environment info
Operating System: OSX

Compiler: gcc

Package used (python/R/jvm/C++): python

`xgboost` version used: 0.6

If installing from source, please provide 

1. The commit hash (`git rev-parse HEAD`) :  9fb46e2c5efbb7ea7bb0cbb0f815dbdc9b720177

**Comment by khotilov** [comment]
In multiclass setting, during each boosting iteration, a tree is built for every class. You have 6 classes, so a single boosting iteration creates 6 trees, and prediction with `ntree_limit=1` returns the results from all these 6 trees. I.e., `ntree_limit` corresponds to the number of iterations in multiclass setting (assuming num_parallel_tree=1)

**Comment by mandarup** [comment]
Thanks, that explains it!

**Comment by mandarup** [comment]
@khotilov  how are the trees arranged?
if `ntree_limit=2` with `6` classes then  are first six trees level-1, then the next six trees level-2? by level I the training sequence.

**Comment by khotilov** [comment]
@mandarup yes, this is correct.
